source/server/worker_impl.cc

```diff
@@ -81,8 +81,20 @@ void WorkerImpl::removeFilterChains(uint64_t listener_tag,
 
 void WorkerImpl::start(GuardDog& guard_dog) {
   ASSERT(!thread_);
-  thread_ =
-      api_.threadFactory().createThread([this, &guard_dog]() -> void { threadRoutine(guard_dog); });
+
+  // In posix, thread names are limited to 15 characters, so contrive to make
+  // sure all interesting data fits there. The naming occurs in
+  // ListenerManagerImpl's constructor: absl::StrCat("worker_", i). Let's say we
+  // have 9999 threads. We'd need, so we need 7 bytes for "worker_", 4 bytes
```

Comment:
9999?  I think three digits should be plenty; two is probably enough.

Reply:
I was being a little future-proof; but there definitely exist machines with #cores getting into 3-digits.